lib/transfer.c

```diff
@@ -144,12 +143,13 @@ CURLcode Curl_fillreadbuffer(struct connectdata *conn, int bytes, int *nreadp)
     }
     return CURLE_OK; /* nothing was read */
   }
-  else if((size_t)nread > buffersize) {
+  else if(res > buffersize) {
```

Comment:
User function `fread_func` could return absolutely incorrect high value, but using only lower 32 bits will completely hide this error.